src/createChainVerifier.ts

```diff
@@ -30,6 +30,32 @@ export function createChainVerifier(caStore: ICaStore): CertificateVerifier {
   };
 }
 
+// returns a function like createChainVerifier which performs host validation on flag
+export function createChainVerifierFactory(hasHostValidation = true) {
```

Comment:
We should modify the original `createChainVerifier` by adding an optional second argument: 

```
interface ICertificateVerifierOpts {
  withHostnameValidation?: boolean
}

export function createChainVerifier(caStore: ICaStore, { withHostnameValidation = true }: ICertificateVerifierOpts) {...}
```